grails-app/services/au/org/ala/spatial/portal/HubWebService.groovy

```diff
@@ -134,35 +137,39 @@ class HubWebService {
 
             HttpClientParams httpParams = client.params
             httpParams.setConnectionManagerTimeout((int) grailsApplication.config.http.timeout)
-
-            if (type == HttpGet.METHOD_NAME) {
-                call = new GetMethod(url)
-            } else {
-                if (type == HttpPut.METHOD_NAME) {
-                    call = new PutMethod(url)
-                } else if (type == HttpPost.METHOD_NAME) {
-                    call = new PostMethod(url)
-
-                    if (nameValues) {
-                        nameValues.each { k, v ->
-                            String key = String.valueOf(k)
-                            String value = String.valueOf(v)
-                            if (key != null && value != null) {
-                                if (v instanceof List) {
-                                    v.each { i ->
-                                        String item = String.valueOf(i)
-                                        if (item) {
-                                            ((PostMethod) call).addParameter(key, item)
-                                        }
-                                    }
-                                } else {
-                                    ((PostMethod) call).addParameter(key, value)
+            List nameValuePairs = new ArrayList();
```

Comment:
the var name `nameValuePairs` suggests a HashMap data structure but you're using a `List`. Best practice is to use `list` in your var name to make this clear when reading code further down. E.g. `nameValuePairList`.